src/hotkey/command_executor.cpp

```diff
@@ -556,7 +556,7 @@ static void event_execute( const SDL_Event& event, command_executor* executor)
 		if(!CKey::is_uncomposable(event.key) && !(mods & KMOD_CTRL) && !(mods & KMOD_ALT) && !(mods & KMOD_GUI)) {
 				return;
 		}
-	} else if(event.type == SDL_TEXTINPUT && strnlen(event.text.text, 32) != 1) {
+	} else if(event.type == SDL_TEXTINPUT && strlen(event.text.text) != 1) {
```

Comment:
Use strnlen. Either upgrade your compiler or write a shim for your personal use.